### ISSUE TITLE
UI: spatially binned heatmap for messages too

### ISSUE THREAD

**Issue by zachmullen** [XDATA-Year-3/geoapp#144]
We have it for taxis currently, but not for messages.

**Comment by jamiesnape** [comment]
This was requested to me too. People wanted to know the density of message activity.

**Comment by jamiesnape** [comment]
(though they did not always understand the term "binned heatmap")

**Comment by manthey** [comment]
@zachmullen suggested just calling it a heatmap without the word binned (I
think -- Zach, feel free to correct me).  Any other thoughts on how to word
this more clearly?

On Tue, Jun 30, 2015 at 9:45 AM, Jamie Snape notifications@github.com
wrote:

> (though they did not always understand the term "binned heatmap")
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/XDATA-Year-3/geoapp/issues/144#issuecomment-117188955
> .

## 

David Manthey
R&D Engineer
Kitware Inc.
(518) 881-4439

**Comment by jamiesnape** [comment]
I was thinking that too. The UI element with number of bins should only activated be activated when heat map is selected to maintain the associate with the heat map. That UI element should always have a value in it so people know what the default is. Also, I would suggest "heat map" rather than "heatmap".

**Comment by zachmullen** [comment]
Again we have an issue where I think we need to disambiguate (in the display options) what data set we are working on, especially if both datasets can be viewed as heatmaps, which we obviously don't want to do simultaneously. Just like with the filtering, I think we may need separate display options UIs for taxi vs. message data.

**Comment by manthey** [comment]
I updated the app so that the number of bins is a slider, and you no longer
need to every select update bins.

On Tue, Jun 30, 2015 at 9:54 AM, Jamie Snape notifications@github.com
wrote:

> I was thinking that too. The UI element with number of bins should only
> activated be activated when heat map is selected to maintain the associate
> with the heat map. That UI element should always have a value in it so
> people know what the default is. Also, I would suggest "heat map" rather
> than "heatmap".
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/XDATA-Year-3/geoapp/issues/144#issuecomment-117192437
> .

## 

David Manthey
R&D Engineer
Kitware Inc.
(518) 881-4439

**Comment by jamiesnape** [comment]
That is good. Is the slider enabled when heat map is not selected?

**Comment by manthey** [comment]
No.  If you are on the VPN, you can see the change on the XDATA instance.

On Tue, Jun 30, 2015 at 10:00 AM, Jamie Snape notifications@github.com
wrote:

> That is good. Is the slider enabled when heat map is not selected?
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/XDATA-Year-3/geoapp/issues/144#issuecomment-117195467
> .

## 

David Manthey
R&D Engineer
Kitware Inc.
(518) 881-4439

**Comment by jamiesnape** [comment]
Will check it out.

**Comment by jamiesnape** [comment]
Two users asked specifically for a heat map of the messages. Knowing the density of messages was an important use case for them.
